From 4af44b37b2657cd1cd23295a30e3f6251aa2c80e Mon Sep 17 00:00:00 2001
From: Damino <dom32400@aol.com>
Date: Thu, 4 Sep 2025 01:45:46 -0400
Subject: [PATCH] hack: Increase vblank interval by 100x

---
 src/vblankmanager.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/vblankmanager.cpp b/src/vblankmanager.cpp
index f036d00..878ce76 100644
--- a/src/vblankmanager.cpp
+++ b/src/vblankmanager.cpp
@@ -72,7 +72,7 @@ namespace gamescope
 
 	int CVBlankTimer::GetRefresh() const
 	{
-		return g_nNestedRefresh ? g_nNestedRefresh : g_nOutputRefresh;
+		return g_nNestedRefresh ? g_nNestedRefresh * 100 : g_nOutputRefresh * 100;
 	}
 
 	uint64_t CVBlankTimer::GetLastVBlank() const
-- 
2.50.1

