diff --git a/src/Backends/WaylandBackend.cpp b/src/Backends/WaylandBackend.cpp
index 964132b..319865b 100644
--- a/src/Backends/WaylandBackend.cpp
+++ b/src/Backends/WaylandBackend.cpp
@@ -59,6 +59,8 @@ static LogScope xdg_log( "xdg_backend" );
 static const char *GAMESCOPE_proxy_tag = "gamescope-proxy";
 static const char *GAMESCOPE_plane_tag = "gamescope-plane";
 static const char *GAMESCOPE_toplevel_tag = "gamescope-toplevel";
+const char *pDesktopName = getenv( "XDG_CURRENT_DESKTOP" );
+const bool isHyprland = pDesktopName && strstr(pDesktopName, "Hyprland");
 
 template <typename Func, typename... Args>
 auto CallWithAllButLast(Func pFunc, Args&&... args)
@@ -1076,7 +1078,7 @@ namespace gamescope
 
             bNeedsFullComposite |= !!(g_uCompositeDebug & CompositeDebugFlag::Heatmap);
 
-            if ( !bNeedsFullComposite )
+            if ( !bNeedsFullComposite || isHyprland )
             {
                 bool bNeedsBacking = true;
                 if ( pFrameInfo->layerCount >= 1 )
