diff --git a/src/Backends/WaylandBackend.cpp b/src/Backends/WaylandBackend.cpp
index 664ed0e..fecd652 100644
--- a/src/Backends/WaylandBackend.cpp
+++ b/src/Backends/WaylandBackend.cpp
@@ -444,6 +444,7 @@ namespace gamescope
         std::atomic<bool> m_bDesiredFullscreenState = { false };
 
         bool m_bHostCompositorIsCurrentlyVRR = false;
+        bool m_bForceVRR = false;
     };
 
     class CWaylandFb final : public CBaseBackendFb
@@ -970,6 +971,9 @@ namespace gamescope
             UpdateFullscreenState();
         }
 
+        const char *forceVrr = getenv( "GSC_FORCE_VRR" );
+        m_bForceVRR = (forceVrr && *forceVrr && atoi(forceVrr) != 0);
+
         UpdateEdid();
         m_pBackend->HackUpdatePatchedEdid();
 
@@ -1136,7 +1140,7 @@ namespace gamescope
     }
     bool CWaylandConnector::IsVRRActive() const
     {
-        return cv_adaptive_sync && m_bHostCompositorIsCurrentlyVRR;
+        return cv_adaptive_sync && (m_bHostCompositorIsCurrentlyVRR || m_bForceVRR);
     }
     std::span<const BackendMode> CWaylandConnector::GetModes() const
     {
