diff --git a/types/wlr_color_management_v1.c b/types/wlr_color_management_v1.c
index aa924e36..22aeea97 100644
--- a/types/wlr_color_management_v1.c
+++ b/types/wlr_color_management_v1.c
@@ -21,6 +21,7 @@ struct wlr_color_management_output_v1 {
 	struct wl_list link;
 
 	struct wl_listener output_destroy;
+	struct wl_listener output_commit;
 };
 
 struct wlr_color_management_surface_v1_state {
@@ -284,6 +285,7 @@ static void cm_output_destroy(struct wlr_color_management_output_v1 *cm_output)
 	}
 	wl_resource_set_user_data(cm_output->resource, NULL); // make inert
 	wl_list_remove(&cm_output->output_destroy.link);
+	wl_list_remove(&cm_output->output_commit.link);
 	wl_list_remove(&cm_output->link);
 	free(cm_output);
 }
@@ -293,6 +295,18 @@ static void cm_output_handle_output_destroy(struct wl_listener *listener, void *
 	cm_output_destroy(cm_output);
 }
 
+static void cm_output_handle_output_commit(struct wl_listener *listener, void *data) {
+	struct wlr_color_management_output_v1 *cm_output =
+		wl_container_of(listener, cm_output, output_commit);
+	const struct wlr_output_event_commit *event = data;
+
+	if (!(event->state->committed & WLR_OUTPUT_STATE_IMAGE_DESCRIPTION)) {
+		return;
+	}
+
+	wp_color_management_output_v1_send_image_description_changed(cm_output->resource);
+}
+
 static void cm_output_handle_resource_destroy(struct wl_resource *resource) {
 	struct wlr_color_management_output_v1 *cm_output = cm_output_from_resource(resource);
 	cm_output_destroy(cm_output);
@@ -745,6 +759,9 @@ static void manager_handle_get_output(struct wl_client *client,
 	cm_output->output_destroy.notify = cm_output_handle_output_destroy;
 	wl_signal_add(&output->events.destroy, &cm_output->output_destroy);
 
+	cm_output->output_commit.notify = cm_output_handle_output_commit;
+	wl_signal_add(&output->events.commit, &cm_output->output_commit);
+
 	wl_list_insert(&manager->outputs, &cm_output->link);
 	wl_resource_set_user_data(cm_output->resource, cm_output);
 }
