diff --git a/types/wlr_color_management_v1.c b/types/wlr_color_management_v1.c
index 18ce6ec2..e5e77c6e 100644
--- a/types/wlr_color_management_v1.c
+++ b/types/wlr_color_management_v1.c
@@ -1051,7 +1051,7 @@ wlr_color_manager_v1_transfer_function_list_from_renderer(struct wlr_renderer *r
 	}
 
 	const enum wp_color_manager_v1_transfer_function list[] = {
-		WP_COLOR_MANAGER_V1_TRANSFER_FUNCTION_SRGB,
+		// WP_COLOR_MANAGER_V1_TRANSFER_FUNCTION_SRGB,
 		WP_COLOR_MANAGER_V1_TRANSFER_FUNCTION_GAMMA22,
 		WP_COLOR_MANAGER_V1_TRANSFER_FUNCTION_ST2084_PQ,
 		WP_COLOR_MANAGER_V1_TRANSFER_FUNCTION_EXT_LINEAR,
