diff --git a/types/wlr_linux_dmabuf_v1.c b/types/wlr_linux_dmabuf_v1.c
index 3165e880..fea2037f 100644
--- a/types/wlr_linux_dmabuf_v1.c
+++ b/types/wlr_linux_dmabuf_v1.c
@@ -8,6 +8,7 @@
 #include <wlr/config.h>
 #include <wlr/interfaces/wlr_buffer.h>
 #include <wlr/render/wlr_renderer.h>
+#include <wlr/render/wlr_texture.h>
 #include <wlr/types/wlr_compositor.h>
 #include <wlr/types/wlr_linux_dmabuf_v1.h>
 #include <wlr/types/wlr_output_layer.h>
@@ -227,6 +228,18 @@ static bool check_import_dmabuf(struct wlr_dmabuf_attributes *attribs, void *dat
 	return true;
 }
 
+static bool check_renderer_dmabuf(struct wlr_dmabuf_attributes *attribs, void *data) {
+	struct wlr_renderer *renderer = data;
+
+	struct wlr_texture *texture = wlr_texture_from_dmabuf(renderer, attribs);
+	if (!texture) {
+		return false;
+	}
+
+	wlr_texture_destroy(texture);
+	return true;
+}
+
 static void params_create_common(struct wl_resource *params_resource,
 		uint32_t buffer_id, int32_t width, int32_t height, uint32_t format,
 		uint32_t flags) {
@@ -1003,6 +1016,12 @@ struct wlr_linux_dmabuf_v1 *wlr_linux_dmabuf_v1_create_with_renderer(struct wl_d
 	struct wlr_linux_dmabuf_v1 *linux_dmabuf =
 		wlr_linux_dmabuf_v1_create(display, version, &feedback);
 	wlr_linux_dmabuf_feedback_v1_finish(&feedback);
+
+	if (linux_dmabuf != NULL) {
+		wlr_linux_dmabuf_v1_set_check_dmabuf_callback(linux_dmabuf,
+			check_renderer_dmabuf, renderer);
+	}
+
 	return linux_dmabuf;
 }
 
